OpenRA.Mods.Common/Traits/SpawnActorOnDeath.cs

```diff
@@ -75,38 +81,43 @@ public void Killed(Actor self, AttackInfo e)
 			if (info.DeathType != null && !e.Damage.DamageTypes.Contains(info.DeathType))
 				return;
 
-			self.World.AddFrameEndTask(w =>
-			{
-				// Actor has been disposed by something else before its death (for example `Enter`).
-				if (self.Disposed)
-					return;
-
-				var td = new TypeDictionary
-				{
-					new ParentActorInit(self),
-					new LocationInit(self.Location),
-					new CenterPositionInit(self.CenterPosition),
-					new FactionInit(faction)
-				};
-
-				if (info.OwnerType == OwnerType.Victim)
-					td.Add(new OwnerInit(self.Owner));
-				else if (info.OwnerType == OwnerType.Killer)
-					td.Add(new OwnerInit(e.Attacker.Owner));
-				else
-					td.Add(new OwnerInit(self.World.Players.First(p => p.InternalName == info.InternalOwner)));
+			// Actor has been disposed by something else before its death (for example `Enter`).
+			if (self.Disposed)
```

Comment:
Moving the execution of all this code from the FrameEndTask to here may have unexpected side-effects.  I don't have the time to check this myself, but another reviewer definitely should.